### DISCUSSION TITLE
Need help in figuring out the cause of flicker on page load causing huge performance drops

### DISCUSSION THREAD

**Discussion by ghost** [gatsbyjs/gatsby#33710] [Help]
Hey all, 
We've been facing an issue with our Gatsby project from quite some time. The webpage loads up and then pauses for a good amount of time and then continues to load up. This tremendously impacts the experience of our users. 
It's been quite confusing to pinpoint the cause of the bottleneck just by looking at the network tab requests.

There is a workaround for the issue which basically toggles the opacity of the body element from 0 to 1 on page load. It fixes the flicker issue but causes a huge drop in pagespeed points (over 30-40 points). 

I've discussed the same issue in the past as well and people seemed to suggest to look at font loading, css imports and stuff. Sadly, at that time I couldn't share the code so it was hard for people to debug. I've now finally been able to share some crucial pieces of code where the imports happen.
 
Link - https://gatsby-flicker-test.netlify.app/
Repo with some important files - https://github1s.com/crio001/gatsby-flicker-test
I had to remove the rest of the files since they are proprietary. I'd be happy to help with any information needed.

Any help would be appreciated.

**Comment by samkit5495** [comment]
did we find any solution for this ?
